### DISCUSSION TITLE
How to install Homebrew on Mac M1 ?

### DISCUSSION THREAD

**Discussion by anurawat01** [Homebrew/discussions#698] [Getting started] [answered]
Can anyone help me to install homebrew I am trying from last two days?

I have tried this link:
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"

But I am getting error:
==> This script will install:
/opt/homebrew/bin/brew
/opt/homebrew/share/doc/homebrew
/opt/homebrew/share/man/man1/brew.1
/opt/homebrew/share/zsh/site-functions/_brew
/opt/homebrew/etc/bash_completion.d/brew
/opt/homebrew

Press RETURN to continue or any other key to abort
==> /usr/bin/sudo /usr/sbin/chown -R anubhav:admin /opt/homebrew
==> Downloading and installing Homebrew...
remote: Enumerating objects: 6, done.
remote: Counting objects: 100% (6/6), done.
remote: Compressing objects: 100% (2/2), done.
remote: Total 7 (delta 4), reused 4 (delta 4), pack-reused 1
Unpacking objects: 100% (7/7), done.
From https://github.com/Homebrew/brew
   f516eded8..51e1b6697  master     -> origin/master
HEAD is now at 51e1b6697 Merge pull request #10527 from reitermarkus/fix-vendor-gems
error: Not a valid ref: refs/remotes/origin/master
fatal: ambiguous argument 'refs/remotes/origin/master': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'

error: RPC failed; curl 56 LibreSSL SSL_read: SSL_ERROR_SYSCALL, errno 54
fatal: the remote end hung up unexpectedly
fatal: early EOF
fatal: index-pack failed
Error: Fetching /opt/homebrew/Library/Taps/homebrew/homebrew-core failed!
fatal: invalid upstream 'origin/master'
Failed during: /opt/homebrew/bin/brew update --force --quiet

**Comment by SMillerDev** [comment, marked as answer]
There's something in your network connection interfering with the SSL connection to github.com (Homebrew's host for package definitions) unfortunately there is not much we can do to help you debug this, but my recommendation would be to try a different network.

**Reply by anurawat01** [reply]
How can I reinstall?

**Reply by SMillerDev** [reply]
To reinstall Homebrew you can just run the installer again and it will try and continue where it failed.

**Reply by anurawat01** [reply]
No, I want to do it from scratch so what should I do?
Because it is creating error again!

**Reply by SMillerDev** [reply]
Reinstalling won't change that. It's a network issue.
